### ISSUE TITLE
Spring Cloud Eureka + Gradle 4.0 doesn't start due to Tomcat child container failure

### ISSUE THREAD

**Issue by wilderrodrigues** [gradle/gradle#2353]
When using Spring Cloud Eureka with Gradle 4.0, the embedded Tomcat fails to start due to a child container failure.

I got it working by doing the following:

1. run ```gradle clean wrapper```
2. run ```./gradlew clean build```
3. run ```./gradlew bootRun```

The ```gradle clean wrapper``` replaces the Gradle 4.0 from the gradlew properties with Gradle 3.3.

### Expected Behavior
The embedded Tomcat should start successfully on port 8080.

### Current Behavior
The embedded Tomcat doesn't start due to the following:

```
14:16:41.447 [QUIET] [system.out] 2017-06-27 14:16:41.446 ERROR 27335 --- [cat-startStop-1] org.apache.catalina.core.ContainerBase   : A child container failed during start
14:16:41.447 [QUIET] [system.out] 
14:16:41.447 [QUIET] [system.out] java.util.concurrent.ExecutionException: org.apache.catalina.LifecycleException: Failed to start component [StandardEngine[Tomcat].StandardHost[localhost].TomcatEmbeddedContext[]]
.
.
.
Caused by: org.gradle.process.internal.ExecException: Process 'command '/Library/Java/JavaVirtualMachines/jdk1.8.0_111.jdk/Contents/Home/bin/java'' finished with non-zero exit value 1
```

### Steps to Reproduce (for bugs)
1. Go to start.apring.io
2. Create a sample project using Eureka Server (check the attached image)
4. Edit the ```build.gradle``` adn edd the snippet below to the end of the file
      ```
      task wrapper(type: Wrapper) {
                gradleVersion = '4.0'
      }```
5. Add the following annotation to your Spring Boot Application main class (probably called DemoApplication.java):
    ```@EnableEurekaServer```
6. run ```gradle wrapper```
7. run ``./gradlew clean build```
8. run ```./gradlew bootRun --debug```

![image](https://user-images.githubusercontent.com/5129209/27587358-61223cda-5b44-11e7-9ce3-05c0c0d4a1f3.png)

### Your Environment
* MacBook Pro macOS Sierra
* Tried with and without Docker 

If this issue should be on the Spring Cloud Eureka, please do not hesitate to let me know.

**Comment by waincent** [comment]
I can run the Spring Cloud Eureka + Gradle 4.0 in the IDEA, but when I build image to run in the docker,  get the same problem like you.

**Comment by wilderrodrigues** [comment]
Could that be that IDEA is running something else (another version) other than 4.0?

Also, try to run it from the command line, with ```./gradlew``` instead of ```gradle```. I think IDEA does not use the wrapper.

Cheers, Wilder

**Comment by waincent** [comment]
@wilderrodrigues 
Your guess is right, when I run the ./gradlew :eureka:bootRun , get the same problem. 

Change gradle to version 3.5, the server is run well.

**Comment by chnyangjie** [comment]
So why eureka can't work with gradle 4.0? It's gradle 4.0's bug or eureka's ?

**Comment by wilderrodrigues** [comment]
I can create an issue with them, with the same content, so both parties can have a look and try to find out.

I would help on that, but a bit full now and will go on holidays soon. If the issue persists, I can help once I'm back.

Cheers, Wilder

**Comment by bmuschko** [comment]
Please open an issue with the Spring Boot team. The issue doesn't look like a Gradle 4.0 issue per se. The Spring Boot plugin might not be compatible with Gradle 4.0 yet. Feel free to reopen this issue if you identify if there's an issue to be addressed in Gradle 4.0.
